### DISCUSSION TITLE
Classroom-Assignment-Teams and Repositories - something is not well thought through here...

### DISCUSSION THREAD

**Discussion by astyve** [community/community#102855] [GitHub Education]
### Hi everyone,

Hi,
I've been using GitHub Classroom for a few years now, but still feel like a novice. 

I have created an **Organization** in which my **classrooms** with their corresponding **assignments** exsists. So far so good. 

When the students accept an assignment, they usually get a repository with a name of the form "name-of-assignment-username". Say if the assignment is called "Assignment 1 find the bug", and the username is "asty", the repository is named "assignment-1-find-the-bug-asty", which, I guess, makes sense.

Similar, if it is a group assignment, the students are asked to give the team/group a name, and that name will be added after the name of the assignment for the repository. In addition a **team** is created in the organisation.
Example: Assignment name = "Assignment 2 Chaosgame", teame name = "Team 4", the GitHub team will be named "Team 4" and the repository will be named "assignment-2-chaosgame-team-4".

So far so good. BUT come next school-year, and I'm going to give the similar assignments but in a new classroom. This time also I want my students to name their teams "Teams 4" etc. But this time it does not work. It turns out that the **organization** already has a team named "Team 4" from last year..!?!? The same problem also relates to the repositories for group/team assignments.

So from this, I have a few questions:

1. Why is the link between GitHub Classroom and the mechanisms like **Classroom** and **Assignments** and **Teams** so poorly mapped to GitHub mechanisms such as **organization**, **Repository** and **Teams**?
2. Now I have created one **organization** for my department at the University, and all assignments etc. via Classroom is in this Organization. Since I have to apply for educational license for each organization I create, I thought this to be the best option. But maybe I'm wrong? Should I have used a new organization for each course for each year/semester?

What do you guys do in terms of Organization vs classes/courses and Classrooms?

Kind Regards,
Arne Styve

**Comment by nimonian** [comment]
Here's what I do here.

I create an organisation for all the master assignments. I create a classroom in that organisation where I configure and create the assignments. I never assign those, and I never add learners to that classroom, they're just the "template assignments".

For every new class, I make a new organisation. When I want to assign to the class, I don't create a new assignment but I reuse the assignment from the "template assignments" classroom.

**Reply by astyve** [reply]
Hm...so you are not really exploiting the possibilities within GitHub Classroom then to give feedback to the students directly in the code through pull-requests etc?
I want to use Classroom the way it was designed to be used (I think): Creating new classrooms for each course/class I teach, adding the roster of students, and the TAs. Then within each class, create assignments that are given to the students, some of which are based on a template repository (which I reuse between classrooms) and some not.
Problem is though, when I create a group-assignment, GitHub Classroom insists on creating a GitHub team for each group. Which is fine, but in creating these teams, the team name does not automatically hold any reference to the assignment or classroom the groups are formed to solve. And hence, I end up with the same group names/team names accross classrooms and name-conflict...

So when you say that "For every new class, I make a new organization. ", you will then have to apply to GitHub to get the educational benefits for all your new organizations you create? And what about within these newly created organizations? Do you there create classroom and assignments that to assign to the students? And if so, have you ever tried creating group assignments?
